### PR TITLE
Added check when removing torrent in periodic cb

### LINKED ISSUE #3350: Error when torrent errors but has been removed already
See the following traceback:

```
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 99, in on_read_data
RuntimeError: [Failure instance: Traceback: <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'stop'
lib\threading.py:801:__bootstrap_inner
lib\threading.py:754:run
lib\site-packages\twisted\_threads\_threadworker.py:46:work
lib\site-packages\twisted\_threads\_team.py:190:doWork
--- <exception caught here> ---
lib\site-packages\twisted\python\threadpool.py:250:inContext
lib\site-packages\twisted\python\threadpool.py:266:<lambda>
lib\site-packages\twisted\python\context.py:122:callWithContext
lib\site-packages\twisted\python\context.py:85:callWithContext
Tribler\Core\APIImplementation\LaunchManyCore.py:573:sesscb_states_callback
]
```

Since the `sesscb_states_callback` is invoked on the threadpool (we should consider changing this so it runs on the reactor thread, which would actually fix this issue), it could be that the download has already been removed on the reactor thread. We should simply add a check to the download states callback to see whether the download exists before calling the `stop` method.

### PR DESCRIPTION
**Pull request by devos50**
Fixes #3350

### CODE REVIEW

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@qstokkink ready for review. I'm pretty sure that some other process is running on the Windows machine (see the GUI screenshots).

File: Tribler/Core/APIImplementation/LaunchManyCore.py [hunk 1]
```diff
@@ -570,8 +570,9 @@ def sesscb_states_callback(self, states_list):
                 new_active_downloads.append(safename)
             elif state == DLSTATUS_STOPPED_ON_ERROR:
                 self._logger.error("Error during download: %s", repr(ds.get_error()))
-                self.downloads.get(tdef.get_infohash()).stop()
-                self.session.notifier.notify(NTFY_TORRENT, NTFY_ERROR, tdef.get_infohash(), repr(ds.get_error()))
+                if tdef.get_infohash() in self.downloads:
```
Comment (qstokkink) [inline Tribler/Core/APIImplementation/LaunchManyCore.py]: Could you use `get_download(tdef.get_infohash())` for this? That method also makes sure the proper locks on the downloads structure are set.

**Comment by devos50** [comment]
@qstokkink this PR should be ready again, just waiting for the GUI requests to finish.

I've downgraded libtorrent to 1.0.7 again and will create some issues for our libtorrent problems.

**Pull request merged**